tests/Util/PhpServicesCreatorTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Tests\Util;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\MakerBundle\Util\PhpServicesCreator;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\PhpFileLoader;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\Finder\Finder;
+
+// For testing parameters conversion
+\define('GLOBAL_CONSTANT', 'value');
+
+class PhpServicesCreatorTest extends TestCase
+{
+    // For testing parameters conversion
+    const CONSTANT_NAME = 1;
+
+    /**
+     * @dataProvider getConfigurationFiles
+     */
+    public function testConvertionYamlServices(string $yamlFile, string $phpFile, string $nameTestCase)
+    {
+        $creator = new PhpServicesCreator();
+        $this->assertSame($phpFile, $creator->convert($yamlFile));
+        /*
+         * test if default comments and configuration are written but ressources can't be loaded.
+         * the container building is tested by the testLoadResources() above.
+         */
+        if ('services_load_ressources' !== $nameTestCase) {
+            $yamlContainerBuilder = new ContainerBuilder();
+            $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source'));
+            $yamlLoader->load($nameTestCase.'.yaml');
+
+            $phpContainerBuilder = new ContainerBuilder();
+            $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/'));
+            $phpLoader->load($nameTestCase.'.php');
+
+            $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+            $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
+            $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+            $this->assertTrue($yamlContainerBuilder->getServiceIds() == $phpContainerBuilder->getServiceIds());
+        }
+    }
+
+    public function getConfigurationFiles()
+    {
+        $finder = Finder::create()->files()->name('*.php')
+            ->in(__DIR__.'/fixtures/YamlConvert/')
+            ->exclude(['FakeClass', 'fake_files', 'container_built_differently', 'source/fixtures_load']);
```

Comment:
I think we can organize these files a bit better:

1) Move the `YamlConvert` directory into `tests/Util` - not `tests/Util/fixtures` - this mixes the `YamlConvert` directory with the fixtures for a different test. And probably rename `YamlConvert` to `yaml_php_convert_fixtures`.

2) Inside the `yaml_php_convert_fixtures`, I would have this structure:

* `source_yaml/`
* `expected_php/`
* any other directories you need

Then you can cleanly use Finder to find all files inside of `source_yaml/` as your test cases, and load the same filename from `expected_php/`. Not need for excluding things.

Reply:
Done.